kernel/src/log_segment.rs

```diff
@@ -469,6 +469,39 @@ impl LogSegment {
         // read the same protocol and metadata schema for both commits and checkpoints
         self.read_actions(engine, schema.clone(), schema, META_PREDICATE.clone())
     }
+
+    /// How many commits since a checkpoint, according to this log segment
+    pub(crate) fn commits_since_checkpoint(&self) -> usize {
+        // we can use 0 as the checkpoint version if there is no checkpoint since `end_version - 0`
+        // is the correct number of commits since a checkpoint if there are no checkpoints
+        let checkpoint_version = self.checkpoint_version.unwrap_or(0);
+        (self.end_version - checkpoint_version)
+            .try_into()
+            .unwrap_or(usize::MAX)
```

Comment:
would it be better to just return a result and let this error?

Reply:
Yeah that makes sense. Wil lupdate